subsys/bluetooth/host/adv.c

```diff
@@ -392,6 +391,124 @@ static int hci_set_ad_ext(struct bt_le_ext_adv *adv, uint16_t hci_op,
 	return bt_hci_cmd_send_sync(hci_op, buf, NULL);
 }
 
+static int hci_set_adv_ext_fragmented(struct bt_le_ext_adv *adv, uint16_t hci_op,
+				      const struct bt_ad *ad, size_t ad_len)
+{
+	struct bt_hci_cp_le_set_ext_adv_data *set_data;
+	struct net_buf *buf;
+	int err;
+
+	for (size_t i = 0; i < ad_len; i++) {
+
+		const struct bt_data *data = ad[i].data;
+
+		for (size_t j = 0; j < ad[i].len; j++) {
+			size_t len = data[j].data_len;
+			uint8_t type = data[j].type;
+			size_t offset = 0;
+
+			/* We can't necessarily set one AD field in a single step. */
+			while (offset < data[j].data_len) {
+				buf = bt_hci_cmd_create(hci_op, sizeof(*set_data));
+				if (!buf) {
+					return -ENOBUFS;
+				}
+
+				set_data = net_buf_add(buf, sizeof(*set_data));
+				(void)memset(set_data, 0, sizeof(*set_data));
+
+				set_data->handle = adv->handle;
+				set_data->frag_pref = BT_HCI_LE_EXT_ADV_FRAG_DISABLED;
```

Comment:
Had to check this up. Maybe we should rename this to ADV_FRAG_MINIMIZE?
Looks strange to disable fragmentation when we are fragmenting the commands.

Reply:
Yes, I agree, this define may be renamed. Do you want me to change it in this PR?

Reply:
hci.h is semi-public, I think a comment should be enough.